packages/react/src/components/Authenticator/Router/Router.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+
+import { useAuthenticator } from '../hooks/useAuthenticator';
+import { ConfirmSignUp } from '../ConfirmSignUp';
+import { ForceNewPassword } from '../ForceNewPassword';
+import { SetupTOTP } from '../SetupTOTP';
+import { SignInSignUpTabs } from '../shared';
+import { ConfirmVerifyUser, VerifyUser } from '../VerifyUser';
+import { ConfirmSignIn } from '../ConfirmSignIn/ConfirmSignIn';
+import { ConfirmResetPassword, ResetPassword } from '../ResetPassword';
+import { isSignInOrSignUpRoute } from '../utils';
+import { RouterProps } from './types';
+
+const getRouteComponent = (route: string) => {
+  switch (route) {
+    case 'authenticated':
+    case 'idle':
+    case 'setup':
+      return () => null;
+    case 'confirmSignUp':
+      return ConfirmSignUp;
+    case 'confirmSignIn':
+      return ConfirmSignIn;
+    case 'setupTOTP':
+      return SetupTOTP;
+    case 'signIn':
+    case 'signUp':
+      return SignInSignUpTabs;
+    case 'forceNewPassword':
+      return ForceNewPassword;
+    case 'resetPassword':
+      return ResetPassword;
+    case 'confirmResetPassword':
+      return ConfirmResetPassword;
+    case 'verifyUser':
+      return VerifyUser;
+    case 'confirmVerifyUser':
+      return ConfirmVerifyUser;
+    default:
+      console.warn(
+        `Unhandled Authenticator route - please open an issue: ${route}`
+      );
+      return () => null;
+  }
+};
+
+export function useRouterChildren(
+  children: RouterProps['children']
+): (props?: Omit<RouterProps, 'children'>) => JSX.Element {
+  const { route, signOut, user } = useAuthenticator(
+    ({ route, signOut, user }) => [route, signOut, user]
+  );
+
+  return React.useMemo(() => {
+    const isUnauthenticatedRoute = !(
+      route === 'authenticated' || route === 'signOut'
+    );
+
+    if (isUnauthenticatedRoute) {
+      return getRouteComponent(route);
+    }
+
+    // `Authenticator` might not have user defined `children` for non SPA use cases.
+    if (!children) {
+      return () => null;
+    }
+
+    return () =>
+      typeof children === 'function'
+        ? children({ signOut, user }) // children is a render prop
+        : children;
+  }, [children, route, signOut, user]);
+}
+
+export function Router({
+  children,
+  className,
+  hideSignUp,
+  variation,
+}: RouterProps) {
+  const { route } = useAuthenticator(({ route }) => [route]);
+  const RouterChildren = useRouterChildren(children);
+
+  return (
+    <RouterChildren
+      className={className}
+      hideSignUp={isSignInOrSignUpRoute(route) ? hideSignUp : undefined}
+      variation={variation}
```

Comment:
This looks good, but I had a question. So with this new design you have to prop drill `className` and `variation` From the `Authenticator` -> `Router` -> `<component i.e. signUp, signIn>` -> `RouteContainer` -> finally to the `View` component so it can be added to the `data-variation` and `className`. 

I suppose that's fine, it's just a pattern we'll have to make sure we remember to do on all future components we add.

Reply:
That is a possibility. The idea of having the route options in the components context is to prevent the need to pass UI specific properties to the state machine and keep a separation of concerns between UI needs and state machine logic